src/cml.js

```diff
@@ -339,7 +340,9 @@ class CML {
 
     if (!rmWatermark) {
       const [, type] = mime.split('/');
+      const sha = await this.triggerSha();
       uri = watermarkUri({ uri, type });
+      uri = addshaUri({ uri, sha });
```

Comment:
I don't think this is the correct place to add this. This just adds a query parameter to urls of published resources.

Reply:
Do we need this as well?